### PR TITLE
Export OutlineDrawingPass from MMDHelper to make it reusable

### PR DESCRIPTION
**Pull request by takahirox**
I exported OutlineDrawingPass `BackFaceCullingOutlinePass` from `MMDHelper`.

Now it can also be used for non-MMD objects like this

image

How to use

```
function init() {

    composer = new THREE.EffectComposer( renderer );
    composer.setSize( width, height );

    renderPass = new THREE.RenderPass( scene, camera );
    composer.addPass( renderPass );

    outlinePass = new THREE.BackFaceCullingOutlinePass( scene, camera );
    composer.addPass( outlinePass );

    copyPass = new THREE.ShaderPass( THREE.CopyShader );
    copyPass.renderToScreen = true;
    composer.addPass( copyPass );

}

function render() {

    composer.render()

}
```

In general this pass is lighter compared to existing `OutlinePass` because
it simply renders only  twice, one is for normal rendering and the other one is for outline drawing.

Reference: https://en.wikipedia.org/wiki/Cel_shading

### CODE REVIEW

**Comment by takahirox** [comment]
Ah, I came to think it should be `OutlineEffect`, not `*Pass` because it isn't a post-processing approach.
Please don't merge this PR until I change that.

**Comment by takahirox** [comment]
Done.

How to use

    function init() {

        renderer = new THREE.WebGLRenderer();
        effect = new THREE.OutlineEffect( renderer );

    }

    function render() {

        effect.render( scene, camera );

    }

File: examples/js/effects/OutlineEffect.js [hunk 1]
```diff
@@ -0,0 +1,296 @@
+/**
+ * @author takahirox / http://github.com/takahirox/
+ *
+ * Reference: https://en.wikipedia.org/wiki/Cel_shading
+ *
+ * // How to set default outline parameters
+ * new THREE.BackFaceCullingOutlinePass( scene, camera, {
```
Comment (mrdoob) [inline examples/js/effects/OutlineEffect.js]: I gues this needs to be updated?

File: examples/webgl_loader_mmd.html [hunk 2]
```diff
@@ -151,7 +153,7 @@
 				camera.aspect = window.innerWidth / window.innerHeight;
 				camera.updateProjectionMatrix();
 
-				renderer.setSize( window.innerWidth, window.innerHeight );
+				helper.setSize( window.innerWidth, window.innerHeight );
```
Comment (mrdoob) [inline examples/webgl_loader_mmd.html]: `helper.setSize()`?

File: examples/webgl_loader_mmd_audio.html [hunk 3]
```diff
@@ -188,7 +190,7 @@
 				camera.aspect = window.innerWidth / window.innerHeight;
 				camera.updateProjectionMatrix();
 
-				renderer.setSize( window.innerWidth, window.innerHeight );
+				helper.setSize( window.innerWidth, window.innerHeight );
```
Comment (mrdoob) [inline examples/webgl_loader_mmd_audio.html]: `helper.setSize()`?

File: examples/webgl_loader_mmd_pose.html [hunk 4]
```diff
@@ -293,7 +295,7 @@
 				camera.aspect = window.innerWidth / window.innerHeight;
 				camera.updateProjectionMatrix();
 
-				renderer.setSize( window.innerWidth, window.innerHeight );
+				helper.setSize( window.innerWidth, window.innerHeight );
```
Comment (mrdoob) [inline examples/webgl_loader_mmd_pose.html]: `helper.setSize()`?

File: examples/js/effects/OutlineEffect.js (hunk 1 above)
Comment (takahirox) [inline examples/js/effects/OutlineEffect.js]: Thanks!

File: examples/js/effects/OutlineEffect.js [hunk 5]
```diff
@@ -261,35 +259,35 @@ THREE.OutlineEffect = function ( renderer, parameters ) {
...
-		this.renderer.setSize( width, height );
+		renderer.setSize( width, height );
 
 	};
 
 	this.render = function ( scene, camera, renderTarget, forceClear ) {
 
-		var tmpAutoClear = this.renderer.autoClear;
-		this.renderer.autoClear = this.autoClear;
+		var tmpAutoClear = renderer.autoClear;
```
Comment (mrdoob) [inline examples/js/effects/OutlineEffect.js]: We usually call this `var currentAutoClear = renderer.autoClear;`.

File: examples/js/effects/OutlineEffect.js [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+
+	this.setSize = function ( width, height ) {
+
+		renderer.setSize( width, height );
+
+	};
+
+	this.render = function ( scene, camera, renderTarget, forceClear ) {
+
+		var tmpAutoClear = renderer.autoClear;
```
Comment (mrdoob) [inline examples/js/effects/OutlineEffect.js]: Minor thing but... we usually call this `var currentAutoClear = renderer.autoClear;`.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**